manifests/addons/gen.sh

```diff
@@ -0,0 +1,83 @@
+#!/usr/bin/env bash
+
+# Copyright Istio Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+WD=$(dirname "$0")
+WD=$(cd "$WD"; pwd)
+
+set -eux
+
+# This script sets up the plain text rendered deployments for addons
+# See samples/addons/README.md for more information
+
+ADDONS="${WD}/../../samples/addons"
+DASHBOARDS="${WD}/../charts/istio-telemetry/grafana/dashboards"
+mkdir -p "${ADDONS}"
+
+# Set up Kiali. The upstream only has an operator, so we will deploy that
+# We set up anonymous authentication as this is for demos
+KIALI_SRC=$(mktemp -d)
+KIALI_VERSION=1.17.0
+pushd "${KIALI_SRC}"
+curl -s -L https://github.com/kiali/kiali-operator/archive/v${KIALI_VERSION}.tar.gz | tar xz
+OPERATOR_ROLE_CREATE="- create" OPERATOR_ROLE_DELETE="- delete" OPERATOR_ROLE_PATCH="- patch" ./kiali-operator-${KIALI_VERSION}/deploy/merge-operator-yaml.sh \
```

Comment:
For the next (v1.18) release, you will not need to set `OPERATOR_ROLE_[CREATE,DELETE,PATCH]` anymore. So when v1.18 comes out (and Istio [moves to it](https://github.com/istio/istio/pull/23577)) you can remove those as [they are set now in the script itself](https://github.com/kiali/kiali-operator/blob/2ce2ba2d3bd40c0e92591986affc1a70bcf99854/deploy/merge-operator-yaml.sh#L123-L126).

In addition, you will need to pass in the `-crc true` option in order to use the `accessible_namespaces: ['**']` that Istio will configure (see below line 46).

Reply:
Awesome, that makes this a lot less hacky. Thanks!

Reply:
@howardjohn You will also need to pass in `--create-operator-namespace false` - I [just added that option](https://github.com/kiali/kiali-operator/pull/31). I realize if we are going to put kiali operator directly in istio-system, that we don't want the all-in-one yaml to attempt to create the namespace and label it as if its owned by the kiali operator.